presto-parquet/src/main/java/io/prestosql/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -192,9 +193,12 @@ public static Domain getDomain(Type type, long rowCount, Statistics<?> statistic
                 return Domain.create(ValueSet.all(type), hasNullValue);
             }
 
-            ParquetIntegerStatistics parquetStatistics = new ParquetIntegerStatistics(
-                    (long) floatToRawIntBits(floatStatistics.getMin()),
-                    (long) floatToRawIntBits(floatStatistics.getMax()));
+            long min = floatToIntBits(floatStatistics.getMin());
+            long max = floatToIntBits(floatStatistics.getMax());
+            if (isFloatingPointNaN(type, min) || isFloatingPointNaN(type, max)) {
```

Comment:
why not just

```suggestion
            if (Float.isNaN(floatStatistics.getMin()) || Float.isNaN(floatStatistics.getMax())) {
```

?